reference-implementation/lib/readable-stream.js

```diff
@@ -700,8 +700,52 @@ class ReadableStreamDefaultReader {
 
     ReadableStreamReaderGenericRelease(this);
   }
+
+  iterator({ cancel = true } = {}) {
+    const O = this;
+    if (typeof O !== 'object') {
+      throw new TypeError();
+    }
+    if (this._ownerReadableStream === undefined) {
+      throw new TypeError();
+    }
+    const iterator = Object.create(ReadableStreamDefaultReaderAsyncIteratorPrototype);
+    iterator._reader = this;
+    iterator._cancel = Boolean(cancel);
+    return iterator;
+  }
+
+  [Symbol.asyncIterator]() {
+    const O = this;
+    return O.iterator();
+  }
 }
 
+const ReadableStreamDefaultReaderAsyncIteratorPrototype = Object.setPrototypeOf({
+  _reader: undefined,
+  _cancel: undefined,
```

Comment:
Why does the prototype have these properties? The prototype is not an instance of ReadableStreamDefaultReaderAsyncIterator.